Use off-the-shelf React Templates for UI

littlealex003 opened the pull request:
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactor

- **What is the current behavior?** (You can also link to an open issue here)
Custom React Material UI Implementation

- **What is the new behavior (if this is a feature change)?**
Off the shelf React Material Chat UI Implementation

D4nte commented on web-chat/src/ChatList.tsx:
```diff
@@ -1,73 +1,38 @@
+    <Message
+      key={currentmessage.timestamp.toString()}
+      authorName={currentmessage.nick}
+      date={currentmessage.timestamp.toLocaleString([], {
```
Maybe best if we extract the formatting of the date in a dedicated function to declutter.

littlealex003 replied:
Done (please review)

D4nte commented on web-chat/src/MessageInput.tsx:
```diff
@@ -1,7 +1,15 @@
-import { Button, Grid, TextField } from '@material-ui/core';
+import { ChangeEvent, KeyboardEvent, useState } from 'react';
 import { useWaku } from './WakuContext';
 
+import {
```
Did the linter add this newline?
```suggestion
import {
```

littlealex003 replied:
I'm guessing so, but it's possible VSCode did; I've been running "npm run fix" on the web-chat directory before I commit, but it should be automated with a pre-commit hook as you mentioned.

D4nte commented on web-chat/src/custom.d.ts:
```diff
@@ -0,0 +1 @@
+declare module '@livechat/ui-kit';
```
To keep it consistent with js-waku, name the file `types.d.ts` and add it to a `types` folder.

You probably also need to update the `web-chat/tsconfig.json` file with

```
 "typeRoots": ["node_modules/@types", "src/types"]
```

like `./tsconfig.json`.

littlealex003 replied:
Ok should be done now, sorry :)

D4nte reviewed (commented):
I must say it looks good and it is an improvement on the current UX so I'd be keen to merge even if it is not the slack/discord feel.

Can you make it so that messages from the same sender are automatically grouped? This should help make it more condensed.

D4nte commented:
`Resolves #86` needs to be in the description, not the title to work. See https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

D4nte commented:
Can we get rid of material-ui with this PR?

littlealex003 commented:
Sure, we can improve the look and feel and customize it and group them,
honestly I didn't realize we were planning on actually using it and was
trying to leave it with as little UI code clutter as possible and looking
as much as it did before by just replacing the custom stuff with the off
the shelf components.

I can group the messages by sender and in general make it a lot smaller,
and get rid of the bubbles; I misunderstood the goal to be more about
making a demo of how to integrate it than a demo we would use.

I can refactor the date function; I did try to remember to run the linter
before every commit; I'll refactor the JS->TS integration code, I was just
doing something to make it work for the demo, I'll try to match what you
have.

On Thu, Apr 29, 2021 at 2:55 AM Franck Royer ***@***.***>
wrote:

> *@D4nte* commented on this pull request.
>
> I must say it looks good and it is an improvement on the current UX so I'd
> be keen to merge even if it is not the slack/discord feel.
>
> Can you make it so that messages from the same sender are automatically
> grouped? This should help make it more condensed.
>
------------------------------
>
> In web-chat/src/ChatList.tsx
> <https://github.com/status-im/js-waku/pull/95#discussion_r622660851>:
>
> >  interface Props {
>
>    messages: ChatMessage[];
>
>  }
>
>
>
>  export default function ChatList(props: Props) {
>
>    const messages = props.messages;
>
>
>
> -  const listItems = messages.map((message) => (
>
> -    <ListItem key={message.timestamp.toString()}>
>
> -      <ListItemText primary={<Message message={message} />} />
>
> -    </ListItem>
>
> +  const listItems = messages.map((currentmessage) => (
>
> +    <Message
>
> +      key={currentmessage.timestamp.toString()}
>
> +      authorName={currentmessage.nick}
>
> +      date={currentmessage.timestamp.toLocaleString([], {
>
>
> Maybe best if we extract the formatting of the date in a dedicated
> function to declutter.
> ------------------------------
>
> In web-chat/src/MessageInput.tsx
> <https://github.com/status-im/js-waku/pull/95#discussion_r622661177>:
>
> >
>
> +import {
>
>
> Did the linter add this newline?
> ⬇️ Suggested change
>
> -
>
> -import {
>
> +import {
>
>
> ------------------------------
>
> In web-chat/src/custom.d.ts
> <https://github.com/status-im/js-waku/pull/95#discussion_r622662877>:
>
> > @@ -0,0 +1 @@
>
> +declare module ***@***.***/ui-kit';
>
>
> To keep it consistent with js-waku, name the file types.d.ts and add it
> to a types folder.
>
> You probably also need to update the web-chat/tsconfig.json file with
>
>  "typeRoots": ***@***.***", "src/types"]
>
>
> like ./tsconfig.json.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/js-waku/pull/95#pullrequestreview-647745512>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP6HJBDN6PRYTDPXCUNSAO3TLCU7FANCNFSM43XO4JDA>
> .
>

littlealex003 commented:
Ah ok, sorry I'm more used to using JIRA than the built in Github issue
system (I'm also more used to using Jenkins than their built in CI)

Honestly I've been using Google cloud git and Gitlab a lot for the last
years.

On Thu, Apr 29, 2021 at 4:35 AM Franck Royer ***@***.***>
wrote:

> Resolves #86 needs to be in the description, not the title to work. See
> https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/js-waku/pull/95#issuecomment-828902495>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP6HJBD5F2EV322ROEVJTELTLDAYDANCNFSM43XO4JDA>
> .
>

littlealex003 commented:
Yes you are right, there are no more dependencies on Material UI, I'll
remove it

On Thu, Apr 29, 2021 at 6:17 AM Franck Royer ***@***.***>
wrote:

> Can we get rid of material-ui with this PR?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/js-waku/pull/95#issuecomment-828928432>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP6HJBBVLAMKDXEUR2TVXELTLDMWLANCNFSM43XO4JDA>
> .
>

D4nte commented:
Thanks for the replies!

Feel free to add a pre-commit hook to lint the code automatically.
Yes, removing bubble and grouping by sender would be great.

Indeed, I hadn't clarify the fact that we were also dogfooding the app (and it took me some times to remember why I wanted it condensed).

Again, the look is much better so I am excited to get this merged with those small tweaks.

littlealex003 commented:
No worries! We are on almost perfectly opposite times...you had plenty of
time to review everything while I slept. First thing in the morning with my
coffee is a good time for me to go over all this stuff.

On Thu, Apr 29, 2021 at 6:49 AM Franck Royer ***@***.***>
wrote:

> Thanks for the replies!
>
> Feel free to add a pre-commit hook to lint the code automatically.
> Yes, removing bubble and grouping by sender would be great.
>
> Indeed, I hadn't clarify the fact that we were also dogfooding the app
> (and it took me some times to remember why I wanted it condensed).
>
> Again, the look is much better so I am excited to get this merged with
> those small tweaks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/js-waku/pull/95#issuecomment-828936882>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP6HJBF2KWKKLF4Y7JT4RA3TLDQNHANCNFSM43XO4JDA>
> .
>

littlealex003 commented:
I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.

Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?

D4nte commented:
> I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.
> 
> Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?

It is fine if you want to get this merged first and then work on getting it more condensed in a second PR.
If that's what you prefer to do then yes, please remove `resolves #86`.

littlealex003 commented:
> > I'll fix this stuff up and squash it into a single commit again, I can see how that's a good best practice to follow.
> > Should I leave "Resolves 86" off until I make it more condensed and just leave this PR as the intermediary refactor from Material to the Chat UI Toolkit?
> 
> It is fine if you want to get this merged first and then work on getting it more condensed in a second PR.
> If that's what you prefer to do then yes, please remove `resolves #86`.

I removed it and squashed everything into a single commit. I think the linter may have reformatted the tsconfig; I can revert that if you prefer, let me know.

I like to do smaller more atomic commits while I'm working but not too hard to squash them all into a single commit at the end.

D4nte commented:
Looks good, I'll take over

D4nte commented:
bors merge

D4nte commented:
bors merge
